src/topology/algebra/module.lean

```diff
@@ -801,3 +806,34 @@ equiv_of_inverse (f₁.prod (f₁.proj_ker_of_right_inverse f₂ h)) (f₂.copro
   (equiv_of_right_inverse f₁ f₂ h).symm y = f₂ y.1 + y.2 := rfl
 
 end continuous_linear_equiv
+
+namespace submodule
+
+variables
+{R : Type*} [ring R]
+{M : Type*} [topological_space M] [add_comm_group M] [module R M]
+{M₂ : Type*} [topological_space M₂] [add_comm_group M₂] [module R M₂]
+
+/-- A submodule `p` is called *complemented* if there exists a continuous projection `M →ₗ[R] p`. -/
+def complemented (p : submodule R M) : Prop := ∃ f : M →L[R] p, ∀ x : p, f x = x
```

Comment:
Being complemented is not a trivial notion in general for modules. So, I expect that it would be useful to have `submodule.complemented` just for the existence of algebraic complements, and then the topological version should have a more specific name. `topo_complemented`?